Adopt more smart pointers under WebCore/page/

cdumez opened the pull request:
#### b2127af72b9226f59303afbbd62011119bdd0fc8
<pre>
Adopt more smart pointers under WebCore/page/
<a href="https://bugs.webkit.org/show_bug.cgi?id=269480">https://bugs.webkit.org/show_bug.cgi?id=269480</a>

Reviewed by Ryosuke Niwa and Brent Fulgham.

Adopt more smart pointers under WebCore/page/, as recommended by the static analyzer.

* Source/WTF/wtf/RunLoop.h:
* Source/WebCore/dom/ElementInlines.h:
(WebCore::Node::protectedParentElement const):
* Source/WebCore/dom/Node.h:
* Source/WebCore/inspector/agents/page/PageDOMDebuggerAgent.cpp:
(WebCore::PageDOMDebuggerAgent::willInsertDOMNode):
* Source/WebCore/page/ContextMenuController.cpp:
(WebCore::prepareContextForQRCode):
* Source/WebCore/page/DragController.cpp:
(WebCore::DragController::tryDocumentDrag):
(WebCore::DragController::prepareForDragStart const):
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::draggedElement):
(WebCore::EventHandler::protectedDraggedElement):
(WebCore::EventHandler::handleMousePressEvent):
(WebCore::EventHandler::scrollRecursively):
(WebCore::EventHandler::logicalScrollRecursively):
(WebCore::EventHandler::subframeForHitTestResult):
(WebCore::EventHandler::handleMouseDoubleClickEvent):
(WebCore::EventHandler::handleMouseMoveEvent):
(WebCore::EventHandler::handleMouseForceEvent):
(WebCore::EventHandler::canDropCurrentlyDraggedImageAsFile const):
(WebCore::EventHandler::dispatchDragEnterOrDragOverEvent):
(WebCore::EventHandler::updateDragAndDrop):
(WebCore::EventHandler::cancelDragAndDrop):
(WebCore::EventHandler::performDragAndDrop):
(WebCore::EventHandler::handleWheelEventInternal):
(WebCore::EventHandler::handleAccessKey):
(WebCore::EventHandler::internalKeyEvent):
(WebCore::EventHandler::didStartDrag):
(WebCore::EventHandler::updateDragStateAfterEditDragIfNeeded):
(WebCore::EventHandler::shouldDispatchEventsToDragSourceElement):
(WebCore::EventHandler::dispatchEventToDragSourceElement):
(WebCore::EventHandler::dispatchDragStartEventOnSourceElement):
(WebCore::EventHandler::handleDrag):
(WebCore::hitTestResultInFrame):
(WebCore::EventHandler::handleTouchEvent):
(WebCore::EventHandler::draggedElement const): Deleted.
* Source/WebCore/page/EventHandler.h:
* Source/WebCore/page/FocusController.cpp:
(WebCore::FocusController::nextFocusableElementOrScopeOwner):
(WebCore::shouldClearSelectionWhenChangingFocusedElement):
* Source/WebCore/page/FrameTree.cpp:
(WebCore::FrameTree::appendChild):
(WebCore::FrameTree::protectedThisFrame const):
(WebCore::FrameTree::scopedChild const):
(WebCore::FrameTree::scopedChildByUniqueName const):
(WebCore::FrameTree::scopedChildBySpecifiedName const):
(WebCore::FrameTree::scopedChildCount const):
(WebCore::FrameTree::find const):
(WebCore::FrameTree::isDescendantOf const):
(WebCore::FrameTree::traverseNext const):
(WebCore::FrameTree::traverseNextSkippingChildren const):
(WebCore::FrameTree::nextAncestorSibling const):
(WebCore::FrameTree::nextRenderedSibling const):
(WebCore::FrameTree::traverseNextRendered const):
(WebCore::FrameTree::deepFirstChild const):
(WebCore::FrameTree::deepLastChild const):
(WebCore::FrameTree::top const):
(WebCore::FrameTree::protectedTop const):
(WebCore::FrameTree::depth const):
(showFrameTree):
* Source/WebCore/page/FrameTree.h:
* Source/WebCore/page/ImageOverlayController.cpp:
(WebCore::ImageOverlayController::selectionQuadsDidChange):
(WebCore::ImageOverlayController::installPageOverlayIfNeeded):
(WebCore::ImageOverlayController::uninstallPageOverlay):
(WebCore::ImageOverlayController::protectedPage const):
* Source/WebCore/page/ImageOverlayController.h:
* Source/WebCore/page/LocalDOMWindow.cpp:
(WebCore::LocalDOMWindow::dispatchAllPendingUnloadEvents):
(WebCore::LocalDOMWindow::canShowModalDialog):
(WebCore::LocalDOMWindow::crypto const):
(WebCore::LocalDOMWindow::processPostMessage):
(WebCore::LocalDOMWindow::focus):
(WebCore::LocalDOMWindow::scrollY const):
(WebCore::LocalDOMWindow::getComputedStyle const):
(WebCore::LocalDOMWindow::getMatchedCSSRules const):
(WebCore::LocalDOMWindow::scrollBy const):
(WebCore::LocalDOMWindow::isSameSecurityOriginAsMainFrame const):
(WebCore::LocalDOMWindow::isAllowedToUseDeviceOrientation const):
(WebCore::LocalDOMWindow::hasPermissionToReceiveDeviceMotionOrOrientationEvents const):
(WebCore::LocalDOMWindow::dispatchLoadEvent):
(WebCore::LocalDOMWindow::removeAllEventListeners):
(WebCore::LocalDOMWindow::setLocation):
(WebCore::LocalDOMWindow::crossDomainAccessErrorMessage):
(WebCore::LocalDOMWindow::isInsecureScriptAccess):
(WebCore::LocalDOMWindow::createWindow):
(WebCore::LocalDOMWindow::open):
(WebCore::LocalDOMWindow::showModalDialog):
(WebCore::LocalDOMWindow::cookieStore):
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::textFragmentIndicatorTimerFired):
(WebCore::LocalFrameView::mapFromLayoutToCSSUnits const):
(WebCore::LocalFrameView::mapFromCSSToLayoutUnits const):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::takeAnyMediaCanStartListener):
(WebCore::Page::setCanStartMedia):
* Source/WebCore/page/PageSerializer.cpp:
(WebCore::PageSerializer::serializeFrame):
* Source/WebCore/page/PerformanceObserver.cpp:
(WebCore::PerformanceObserver::protectedPerformance const):
(WebCore::PerformanceObserver::observe):
(WebCore::PerformanceObserver::disconnect):
* Source/WebCore/page/PerformanceObserver.h:
* Source/WebCore/page/PrintContext.cpp:
(WebCore::PrintContext::spoolPage):
* Source/WebCore/page/ResizeObserver.cpp:
(WebCore::ResizeObserver::isReachableFromOpaqueRoots const):
* Source/WebCore/page/SpatialNavigation.cpp:
(WebCore::scrollInDirection):
* Source/WebCore/page/TextIndicator.h:
(WebCore::TextIndicator::protectedContentImage const):
* Source/WebCore/page/cocoa/WebTextIndicatorLayer.mm:
(createBounceAnimation):
(createContentCrossfadeAnimation):
* Source/WebCore/page/mac/EventHandlerMac.mm:
(WebCore::EventHandler::passSubframeEventToSubframe):
* Source/WebCore/page/mac/ImageOverlayControllerMac.mm:
(WebCore::ImageOverlayController::updateDataDetectorHighlights):
(WebCore::ImageOverlayController::handleDataDetectorAction):
* Source/WebCore/page/scrolling/ScrollingTreeScrollingNodeDelegate.h:
(WebCore::ScrollingTreeScrollingNodeDelegate::protectedScrollingNode const):
* Source/WebCore/page/scrolling/ThreadedScrollingTreeScrollingNodeDelegate.cpp:
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::updateSnapScrollState):
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::updateUserScrollInProgressForEvent):
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::createTimer):
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::startAnimationCallback):
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::stopAnimationCallback):
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::immediateScrollBy):
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::pageScaleFactor const):
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::willStartAnimatedScroll):
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::didStopAnimatedScroll):
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::willStartWheelEventScroll):
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::didStopWheelEventScroll):
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::willStartScrollSnapAnimation):
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::didStopScrollSnapAnimation):
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::scrollExtents const):

Canonical link: <a href="https://commits.webkit.org/274827@main">https://commits.webkit.org/274827@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9ba49e4875a5229b852870e93e41046ac80d8ecb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/40052 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/19064 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/42508 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/42597 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/36027 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/42359 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/22058 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/16393 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/33334 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/40626 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/16118 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/34657 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/13887 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/13989 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/35611 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/43875 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/33505 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/36401 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/35950 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/39646 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/39678 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/14867 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/12203 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/37913 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/16486 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/46686 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/16535 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/9604 "Passed tests") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5299 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/16130 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebCore/page/EventHandler.cpp:
```diff
@@ -4239,7 +4244,7 @@ bool EventHandler::handleDrag(const MouseEventWithHitTestResults& event, CheckDr
     
     // Careful that the drag starting logic stays in sync with eventMayStartDrag().
-    if (m_mouseDownMayStartDrag && !dragState().source) {
+    if (m_mouseDownMayStartDrag && !draggedElement()) {
```
I'm a little confused about why we don't want to hold a single `protectedDragElement()` during the scope of the `handleDrag()` method in its entirety, rather than repeatedly calling `draggedElement()`.

Does the result of `draggedElement()` change as we pass through this code?

brentfulgham commented on Source/WebCore/page/EventHandler.cpp:
```diff
@@ -4320,19 +4325,19 @@ bool EventHandler::handleDrag(const MouseEventWithHitTestResults& event, CheckDr
-        if (dragState().source && dragState().type == DragSourceAction::DHTML && !dragState().dataTransfer->hasDragImage()) {
-            dragState().source->document().updateStyleIfNeeded();
-            if (auto* renderer = dragState().source->renderer()) {
+        if (draggedElement() && dragState().type == DragSourceAction::DHTML && !dragState().dataTransfer->hasDragImage()) {
```
Or at the very least, in a block like this where we check for `draggedElement()` then do things with it, why not get a `protectedDragElement()` here and hold it while inside this block (and perhaps `WTFMove()` into `setDragImage`?)

cdumez replied:
I'll look. We may indeed benefit by storing the dragElement in a variable .

cdumez replied:
I did the WTFMove()

brentfulgham commented on Source/WebCore/page/EventHandler.cpp:
```diff
@@ -5022,14 +5027,14 @@ HandleUserInputEventResult EventHandler::handleTouchEvent(const PlatformTouchEve
             cancelEvent.setTouchPoints(cancelEventPoints);
-            document.page()->pointerCaptureController().dispatchEventForTouchAtIndex(
-                *touchTarget, cancelEvent, index, !index, *document.windowProxy(), { 0, 0 });
+            document->page()->pointerCaptureController().dispatchEventForTouchAtIndex(
```
Should `page()` be protected here? I guess the `Ref` on `document` might already do that.

cdumez replied:
Yes it should. I didn't go through ALL the static analyzer errors in this folder yet to reduce patch size. That said, since I am touching this line, I may as well fix it.

brentfulgham reviewed (approved):
Looks good. r=me

cdumez commented on Source/WebCore/page/EventHandler.cpp:
```diff
@@ -4249,7 +4254,7 @@ bool EventHandler::handleDrag(const MouseEventWithHitTestResults& event, CheckDr
         if (RefPtr page = frame->page())
             dragState().source = page->dragController().draggableElement(frame.ptr(), result.protectedTargetElement().get(), m_mouseDownContentsPosition, dragState());
```
Oh I remember now, the dragged element gets updated here.

cdumez committed:
Adopt more smart pointers under WebCore/page/
https://bugs.webkit.org/show_bug.cgi?id=269480

Reviewed by Ryosuke Niwa and Brent Fulgham.

Adopt more smart pointers under WebCore/page/, as recommended by the static analyzer.

* Source/WTF/wtf/RunLoop.h:
* Source/WebCore/dom/ElementInlines.h:
(WebCore::Node::protectedParentElement const):
* Source/WebCore/dom/Node.h:
* Source/WebCore/inspector/agents/page/PageDOMDebuggerAgent.cpp:
(WebCore::PageDOMDebuggerAgent::willInsertDOMNode):
* Source/WebCore/page/ContextMenuController.cpp:
(WebCore::prepareContextForQRCode):
* Source/WebCore/page/DragController.cpp:
(WebCore::DragController::tryDocumentDrag):
(WebCore::DragController::prepareForDragStart const):
* Source/WebCore/page/EventHandler.cpp:
(WebCore::EventHandler::draggedElement):
(WebCore::EventHandler::protectedDraggedElement):
(WebCore::EventHandler::handleMousePressEvent):
(WebCore::EventHandler::scrollRecursively):
(WebCore::EventHandler::logicalScrollRecursively):
(WebCore::EventHandler::subframeForHitTestResult):
(WebCore::EventHandler::handleMouseDoubleClickEvent):
(WebCore::EventHandler::handleMouseMoveEvent):
(WebCore::EventHandler::handleMouseForceEvent):
(WebCore::EventHandler::canDropCurrentlyDraggedImageAsFile const):
(WebCore::EventHandler::dispatchDragEnterOrDragOverEvent):
(WebCore::EventHandler::updateDragAndDrop):
(WebCore::EventHandler::cancelDragAndDrop):
(WebCore::EventHandler::performDragAndDrop):
(WebCore::EventHandler::handleWheelEventInternal):
(WebCore::EventHandler::handleAccessKey):
(WebCore::EventHandler::internalKeyEvent):
(WebCore::EventHandler::didStartDrag):
(WebCore::EventHandler::updateDragStateAfterEditDragIfNeeded):
(WebCore::EventHandler::shouldDispatchEventsToDragSourceElement):
(WebCore::EventHandler::dispatchEventToDragSourceElement):
(WebCore::EventHandler::dispatchDragStartEventOnSourceElement):
(WebCore::EventHandler::handleDrag):
(WebCore::hitTestResultInFrame):
(WebCore::EventHandler::handleTouchEvent):
(WebCore::EventHandler::draggedElement const): Deleted.
* Source/WebCore/page/EventHandler.h:
* Source/WebCore/page/FocusController.cpp:
(WebCore::FocusController::nextFocusableElementOrScopeOwner):
(WebCore::shouldClearSelectionWhenChangingFocusedElement):
* Source/WebCore/page/FrameTree.cpp:
(WebCore::FrameTree::appendChild):
(WebCore::FrameTree::protectedThisFrame const):
(WebCore::FrameTree::scopedChild const):
(WebCore::FrameTree::scopedChildByUniqueName const):
(WebCore::FrameTree::scopedChildBySpecifiedName const):
(WebCore::FrameTree::scopedChildCount const):
(WebCore::FrameTree::find const):
(WebCore::FrameTree::isDescendantOf const):
(WebCore::FrameTree::traverseNext const):
(WebCore::FrameTree::traverseNextSkippingChildren const):
(WebCore::FrameTree::nextAncestorSibling const):
(WebCore::FrameTree::nextRenderedSibling const):
(WebCore::FrameTree::traverseNextRendered const):
(WebCore::FrameTree::deepFirstChild const):
(WebCore::FrameTree::deepLastChild const):
(WebCore::FrameTree::top const):
(WebCore::FrameTree::protectedTop const):
(WebCore::FrameTree::depth const):
(showFrameTree):
* Source/WebCore/page/FrameTree.h:
* Source/WebCore/page/ImageOverlayController.cpp:
(WebCore::ImageOverlayController::selectionQuadsDidChange):
(WebCore::ImageOverlayController::installPageOverlayIfNeeded):
(WebCore::ImageOverlayController::uninstallPageOverlay):
(WebCore::ImageOverlayController::protectedPage const):
* Source/WebCore/page/ImageOverlayController.h:
* Source/WebCore/page/LocalDOMWindow.cpp:
(WebCore::LocalDOMWindow::dispatchAllPendingUnloadEvents):
(WebCore::LocalDOMWindow::canShowModalDialog):
(WebCore::LocalDOMWindow::crypto const):
(WebCore::LocalDOMWindow::processPostMessage):
(WebCore::LocalDOMWindow::focus):
(WebCore::LocalDOMWindow::scrollY const):
(WebCore::LocalDOMWindow::getComputedStyle const):
(WebCore::LocalDOMWindow::getMatchedCSSRules const):
(WebCore::LocalDOMWindow::scrollBy const):
(WebCore::LocalDOMWindow::isSameSecurityOriginAsMainFrame const):
(WebCore::LocalDOMWindow::isAllowedToUseDeviceOrientation const):
(WebCore::LocalDOMWindow::hasPermissionToReceiveDeviceMotionOrOrientationEvents const):
(WebCore::LocalDOMWindow::dispatchLoadEvent):
(WebCore::LocalDOMWindow::removeAllEventListeners):
(WebCore::LocalDOMWindow::setLocation):
(WebCore::LocalDOMWindow::crossDomainAccessErrorMessage):
(WebCore::LocalDOMWindow::isInsecureScriptAccess):
(WebCore::LocalDOMWindow::createWindow):
(WebCore::LocalDOMWindow::open):
(WebCore::LocalDOMWindow::showModalDialog):
(WebCore::LocalDOMWindow::cookieStore):
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::textFragmentIndicatorTimerFired):
(WebCore::LocalFrameView::mapFromLayoutToCSSUnits const):
(WebCore::LocalFrameView::mapFromCSSToLayoutUnits const):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::takeAnyMediaCanStartListener):
(WebCore::Page::setCanStartMedia):
* Source/WebCore/page/PageSerializer.cpp:
(WebCore::PageSerializer::serializeFrame):
* Source/WebCore/page/PerformanceObserver.cpp:
(WebCore::PerformanceObserver::protectedPerformance const):
(WebCore::PerformanceObserver::observe):
(WebCore::PerformanceObserver::disconnect):
* Source/WebCore/page/PerformanceObserver.h:
* Source/WebCore/page/PrintContext.cpp:
(WebCore::PrintContext::spoolPage):
* Source/WebCore/page/ResizeObserver.cpp:
(WebCore::ResizeObserver::isReachableFromOpaqueRoots const):
* Source/WebCore/page/SpatialNavigation.cpp:
(WebCore::scrollInDirection):
* Source/WebCore/page/TextIndicator.h:
(WebCore::TextIndicator::protectedContentImage const):
* Source/WebCore/page/cocoa/WebTextIndicatorLayer.mm:
(createBounceAnimation):
(createContentCrossfadeAnimation):
* Source/WebCore/page/mac/EventHandlerMac.mm:
(WebCore::EventHandler::passSubframeEventToSubframe):
* Source/WebCore/page/mac/ImageOverlayControllerMac.mm:
(WebCore::ImageOverlayController::updateDataDetectorHighlights):
(WebCore::ImageOverlayController::handleDataDetectorAction):
* Source/WebCore/page/scrolling/ScrollingTreeScrollingNodeDelegate.h:
(WebCore::ScrollingTreeScrollingNodeDelegate::protectedScrollingNode const):
* Source/WebCore/page/scrolling/ThreadedScrollingTreeScrollingNodeDelegate.cpp:
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::updateSnapScrollState):
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::updateUserScrollInProgressForEvent):
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::createTimer):
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::startAnimationCallback):
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::stopAnimationCallback):
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::immediateScrollBy):
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::pageScaleFactor const):
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::willStartAnimatedScroll):
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::didStopAnimatedScroll):
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::willStartWheelEventScroll):
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::didStopWheelEventScroll):
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::willStartScrollSnapAnimation):
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::didStopScrollSnapAnimation):
(WebCore::ThreadedScrollingTreeScrollingNodeDelegate::scrollExtents const):

Canonical link: https://commits.webkit.org/274827@main
